solutions/gold/cses-1665.mdx

```diff
@@ -10,405 +10,70 @@ as $(\text{Skill of last person}) - (\text{Skill of first person})$. Let $s_i$
-	return a / b + ((a ^ b) > 0 && a % b);
-}  // divide a by b rounded up
-ll fdiv(ll a, ll b) {
-	return a / b - ((a ^ b) < 0 && a % b);
-}  // divide a by b rounded down
-
-tcT > bool ckmin(T &a, const T &b) {
-	return b < a ? a = b, 1 : 0;
-}  // set a = min(a,b)
-tcT > bool ckmax(T &a, const T &b) { return a < b ? a = b, 1 : 0; }
-
-tcTU > T fstTrue(T lo, T hi, U f) {
-	hi++;
-	assert(lo <= hi);  // assuming f is increasing
-	while (lo < hi) {  // find first index such that f is true
-		T mid = lo + (hi - lo) / 2;
-		f(mid) ? hi = mid : lo = mid + 1;
-	}
-	return lo;
-}
-tcTU > T lstTrue(T lo, T hi, U f) {
-	lo--;
-	assert(lo <= hi);  // assuming f is decreasing
-	while (lo < hi) {  // find first index such that f is true
-		T mid = lo + (hi - lo + 1) / 2;
-		f(mid) ? lo = mid : hi = mid - 1;
-	}
-	return lo;
-}
-tcT > void remDup(vector<T> &v) {  // sort and remove duplicates
-	sort(all(v));
-	v.erase(unique(all(v)), end(v));
-}
-tcTU > void erase(T &t, const U &u) {  // don't erase
-	auto it = t.find(u);
-	assert(it != end(t));
-	t.erase(it);
-}  // element that doesn't exist from (multi)set
-
-// INPUT
-#define tcTUU tcT, class... U
-tcT > void re(complex<T> &c);
-tcTU > void re(pair<T, U> &p);
-tcT > void re(V<T> &v);
-tcT, size_t SZ > void re(AR<T, SZ> &a);
-
-tcT > void re(T &x) { cin >> x; }
-void re(db &d) {
-	str t;
-	re(t);
-	d = stod(t);
-}
-void re(ld &d) {
-	str t;
-	re(t);
-	d = stold(t);
-}
-tcTUU > void re(T &t, U &...u) {
-	re(t);
-	re(u...);
-}
-
-tcT > void re(complex<T> &c) {
-	T a, b;
-	re(a, b);
-	c = {a, b};
-}
-tcTU > void re(pair<T, U> &p) { re(p.f, p.s); }
-tcT > void re(V<T> &x) { trav(a, x) re(a); }
-tcT, size_t SZ > void re(AR<T, SZ> &x) { trav(a, x) re(a); }
-tcT > void rv(int n, V<T> &x) {
-	x.rsz(n);
-	re(x);
-}
-
-// TO_STRING
-#define ts to_string
-str ts(char c) { return str(1, c); }
-str ts(const char *s) { return (str)s; }
-str ts(str s) { return s; }
-str ts(bool b) {
-#ifdef LOCAL
-	return b ? "true" : "false";
-#else
-	return ts((int)b);
-#endif
-}
-tcT > str ts(complex<T> c) {
-	stringstream ss;
-	ss << c;
-	return ss.str();
-}
-str ts(V<bool> v) {
-	str res = "{";
-	F0R(i, sz(v)) res += char('0' + v[i]);
-	res += "}";
-	return res;
-}
-template <size_t SZ> str ts(bitset<SZ> b) {
-	str res = "";
-	F0R(i, SZ) res += char('0' + b[i]);
-	return res;
-}
-tcTU > str ts(pair<T, U> p);
-tcT > str ts(T v) {  // containers with begin(), end()
-#ifdef LOCAL
-	bool fst = 1;
-	str res = "{";
-	for (const auto &x : v) {
-		if (!fst) res += ", ";
-		fst = 0;
-		res += ts(x);
+const int N = 101, X = 5e3 + 1, M = 1e9 + 7,
+          K = 5e3;  // K is offset to account for negatives
+int n, x, s[N];     // x -> max penalty
+// dp[N][N][X] -> dp[i][j][k] = first i people, j unfinished groups, k penalty
+// for convenience, we 1-index dp array and 0-index person array so dp[i][j][k]
+// lines up with s[i] (i.e. skill of person i + 1 = s[i])
+int dp1[N][X + K], dp2[N][X + K];  // dp1 -> i, dp2 -> i + 1
```

Comment:
can you just use vectors instead

Reply:
why vectors?

Reply:
easier to debug

Reply:
bump

Reply:
i updated it to use vectors, but i dont use them very often so im not sure if there's any standard practices im neglecting

Reply:
nah u did good